km/km_snapshot.c

```diff
@@ -520,6 +521,10 @@ int km_snapshot_create(km_vcpu_t* vcpu, char* label, char* description, char* du
       km_warnx("Cannot create snapshot with GDB running");
       return -EBUSY;
    }
+   if (km_have_forked() != 0) {
```

Comment:
From what I understand this gets wiped out on exec.

Reply:
yes, I need to add it to the state that is saved across exec.

Reply:
I think km_called_via_exec can be reused for this purpose.

Reply:
If you are running a program that does not fork but does exec to another program then no new process is created and we should not see any cases of ambiguous half open pipes/socketpairs.
I think we need to propagate the fork count in exec.